src/oom/oomd-manager.c

```diff
@@ -454,8 +455,30 @@ static int monitor_memory_pressure_contexts_handler(sd_event_source *s, uint64_t
                                 return 0;
                         }
                 }
+        } else {
+                /* If any monitored cgroup is within MEM_PRESSURE_INTERVAL_USEC + 1s second of hitting their pressure
+                 * limit, start getting kill candidates. This saves CPU cycles from doing it every interval. */
+                OomdCGroupContext *c;
+                HASHMAP_FOREACH(c, m->monitored_mem_pressure_cgroup_contexts) {
+                        if ((now(CLOCK_MONOTONIC) - c->last_hit_mem_pressure_limit) > (MEM_PRESSURE_INTERVAL_USEC + 1 * USEC_PER_SEC))
+                                continue;
+
+                        r = update_monitored_cgroup_contexts_candidates(
+                                        m->monitored_mem_pressure_cgroup_contexts, &m->monitored_mem_pressure_cgroup_contexts_candidates);
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0)
+                                log_debug_errno(r, "Failed to update monitored memory pressure candidate cgroup contexts, ignoring: %m");
+                        else
+                                got_candidates = true;
+                }
         }
 
+
+candidate_cleanup:
```

Comment:
There's lots of return points in this function that won't go through this - would that be a problem? Would having a _cleanup_ decorated pointer that gets initialized where you currently set the bool to true be cleaer as a solution?

Reply:
that's a great idea